ui/src/app/views/new-dataset-modal/component.tsx

```diff
@@ -59,53 +96,114 @@ class NewDataSetModal extends React.Component<Props, State> {
       }
     }
   }
 
+  changeExportToNewNotebook() {
+    if (this.state.exportToNewNotebook) {
+      this.setState({exportToNewNotebook: false});
+    } else {
+      this.setState({exportToNewNotebook: true});
+    }
+  }
+
+  async generateQuery() {
+    const {workspaceNamespace, workspaceId} = this.props;
+    const dataSet: DataSetRequest = {
+      name: '',
+      conceptSetIds: this.props.selectedConceptSetIds,
+      cohortIds: this.props.selectedCohortIds,
+      values: this.props.selectedValues,
+      includesAllParticipants: this.props.includesAllParticipants,
+    };
+    const sqlQueries = await dataSetApi().generateQuery(workspaceNamespace, workspaceId, dataSet);
+    this.setState({queries: sqlQueries.queryList});
+  }
+
   render() {
-    const {name, nameTouched, conflictDataSetName, missingDataSetInfo} = this.state;
+    const {
+      conflictDataSetName,
+      exportToNewNotebook,
+      loading,
+      missingDataSetInfo,
+      name,
+      notebookName,
+      notebooksLoading,
+      existingNotebooks,
+      queries,
+    } = this.state;
 
-    const errors = validate({name}, {
+    const errors = validate({name, notebookName}, {
       name: {
         presence: {allowEmpty: false}
+      },
+      notebookName: {
+        presence: {allowEmpty: !exportToNewNotebook},
+        exclusion: {
+          within: existingNotebooks.map(fd => fd.name.slice(0, -6)),
+          message: 'already exists'
+        }
       }
     });
-    return <Modal>
+    return <Modal loading={loading}>
       <ModalTitle>Save Dataset</ModalTitle>
       <ModalBody>
         <div>
-          <ValidationError>
-            {summarizeErrors(nameTouched && errors && errors.name)}
-          </ValidationError>
           {conflictDataSetName &&
           <AlertDanger>DataSet with same name exist</AlertDanger>
           }
           {missingDataSetInfo &&
           <AlertDanger> Data state cannot save as some information is missing</AlertDanger>
           }
-          <TextInput type='text' autoFocus placeholder='Dataset Name'
+          <TextInput type='text' autoFocus placeholder='Data Set Name'
                      value={name}
                      onChange={v => this.setState({
-                       name: v, nameTouched: true,
-                       conflictDataSetName: false
+                       name: v, conflictDataSetName: false
                      })}/>
         </div>
+        <div style={{display: 'flex', alignItems: 'center', marginTop: '1rem'}}>
+          <CheckBox style={{height: 17, width: 17}}
+                    onChange={() => this.changeExportToNewNotebook()} />
+          <div style={{marginLeft: '.5rem', color: 'black'}}>Export to new Python notebook</div>
+        </div>
+        {exportToNewNotebook && <React.Fragment>
+          {(queries.length === 0 || notebooksLoading) && <SpinnerOverlay />}
+          <TextArea disabled={true} onChange={() => {}} style={{marginTop: '1rem'}}
```

Comment:
Question: Why do we have to explicitly mention onChange?

Reply:
It is a required field for TextArea